elastic/kibana
# [Fleet] Fix agents count in agent list table and add tooltip with correct info

Fixes https://github.com/elastic/kibana/issues/195441

## Summary
Selection agent count on agent list table gets incorrect when there are multiple hosted agents, especially if they are on inactive state. In fact to calculate the selected number of agents we were getting hosted agents, but without taking into account the filtering applied on the page, i.e. we were always getting all the hosted agent (inactive too). 
This caused the final calculation to be off. 

In this PR I'm fixing [the query](https://github.com/elastic/kibana/pull/197834/files#diff-9707a4b93a96749876e4cf173a0b39cd5a620e311e2652c5ed4b8670ca7e6becR309-R320) used to get those agents to take in account the filters and I'm also adding a small tooltip that breaks up the number of agents (selected, total, hosted)

### Testing
- Make sure to have many agents, hosted and not in different states (inactive, unenrolled)
- To make a hosted agent inactive follow the steps explained [here](https://github.com/elastic/kibana/issues/195441)
- Verify that the selection numbers are correct: select agents on all pages and hover on the new tooltip shown besides the "selected agents". This number should match the number shown on the actions dropdown


### Screenshots
<img width="2376" alt="Screenshot 2024-10-25 at 17 00 44" src="https://github.com/user-attachments/assets/81d2836a-f997-4ccb-a23c-3d2cfbfa62d3">
<img width="2409" alt="Screenshot 2024-10-25 at 17 00 59" src="https://github.com/user-attachments/assets/ba21933a-f1e6-457e-8059-e87b3e29a7d1">


https://github.com/user-attachments/assets/c153c491-29a1-481c-a3e3-25bab6412963




### Checklist
